lib/smack/SmackRep.cpp

```diff
@@ -653,32 +807,46 @@ unsigned SmackRep::numElements(const llvm::Constant* v) {
 }
 
 void SmackRep::addInit(unsigned region, const llvm::Value* addr, const llvm::Constant* val) {
-  addInit(region, expr(addr), val);
+  if (const llvm::GlobalValue* V = dyn_cast<const GlobalValue>(addr)) { 
+	if (SmackOptions::InferFieldOverlap)
+	  addInit(region, expr(addr), val, V, isFieldDisjoint(V, 0));
+	else
+	  addInit(region, expr(addr), val, V, false);
+  }
+  else
+	  assert(0 && "addInit() should initialize global values?");
 }
 
-void SmackRep::addInit(unsigned region, const Expr* addr, const llvm::Constant* val) {
+void SmackRep::addInit(unsigned region, const Expr* addr, const llvm::Constant* val, const llvm::GlobalValue* V, bool safety) {
   using namespace llvm;
 
   if (isInt(val)) {
-    staticInits.push_back( Stmt::assign(mem(region,addr), expr(val)) );
+    staticInits.push_back( SmackOptions::BitVectors? (safety? Stmt::assign(mem(region, addr, getIntSize(val)), expr(val)) : storeAsBytes(region, getIntSize(val), addr, expr(val))) : Stmt::assign(mem(region,addr), expr(val)) );
```

Comment:
The case-splitting on `SmackOptions::BitVectors` for store assignments in this method should be refactored into a separate method, similarly to `SmackRep::store`.